modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -12,6 +12,10 @@ import Tools from '../util/Tools';
 import Node from './Node';
 import SaxParser from './SaxParser';
 import Schema from './Schema';
+import { BlobCache } from '../file/BlobCache';
+import { uniqueId } from '../../file/ImageScanner';
+import * as Conversions from '../../file/Conversions';
+import { buildBase64DataUri } from 'tinymce/core/html/Base64Uris';
```

Comment:
This should be a relative import path, instead of an alias path.

- [ ] Fix import path

Reply:
Why are aliases bad? I made sure they were configured in every `tsconfig.json` so they can be used where it makes sense 🤔

Reply:
Because if you try and import this from somewhere else then it won't work, as TypeScript doesn't convert them to relative paths when it generates the JS, which leads to broken import paths.

Reply:
Additionally, if we were to distribute the types, this would fail for anyone trying to use them since they'd also have to setup the aliases. So really aliases should really only be used in tests or the like, which aren't something that'd be packaged or consumed elsewhere.

Reply:
Wasn't that only a problem when we published libraries, and even then not in the top-level editor project that doesn't publish as a library? It certainly shouldn't matter in the monorepo?

I admit I haven't checked, so if the aliases are pointless let's delete them so this becomes a compile error instead of a social concept. That will make the tests more annoying though.

Reply:
As I said, it'll be an issue if we ever want to publish the types so we really shouldn't be doing it. You are right though that for tinymce specifically we only distribute the bundled JS, so the aliases aren't really an issue there (with the exception of harbinger).

I'd also be against removing the aliases, as otherwise in tests you just end up with a massive `../../../../../../` path since you need to traverse up a lot of directories. Most IDE's have a setting that allows you to only use the alias if it's outside the configured tsconfig paths so I'd suggest we make sure to configure that instead (not sure if VS code has that). As an example, for IntelliJ we have this setting:
![image](https://user-images.githubusercontent.com/1575550/75831928-9de0fc80-5e00-11ea-91dd-c7da7191094f.png)

Reply:
I think there was some discussion at one stage about seeing if we could right a linter rule to stop these being used in inappropriate places. Maybe we should see if we can prioritise that? 🤷‍♀

Reply:
I think we should blanket allow them in `modules/tinymce` - types we probably need a script processed solution for anyway - and find a way to disallow them in other parts of the monorepo (except for tests). We rely on aliases to, for example, map core references to `tinymce` global references in plugins.

https://github.com/tinymce/tinymce/blob/develop/modules/tinymce/src/plugins/advlist/main/ts/Plugin.ts#L8

Either way we definitely need lint rules to enforce code style decisions like this. Also perhaps IDE configuration, if vscode does has a setting I'll resurrect my branch that adds a vscode workspace file and encourage the team to use it. We've made great use of that in RTC.

Reply:
I added a lint rules for general libraries a while back was harder to do for tiny so I didn't do that maybe it can be expanded. https://github.com/tinymce/tslint-rules/blob/master/src/test/rules/no-path-alias-imports/src/main/test.ts.lint